src/QuartoNotebookWorker/src/refresh.jl

```diff
@@ -1,8 +1,31 @@
 function refresh!(path, original_options, options = original_options)
-    # Current directory should always start out as the directory of the
-    # notebook file, which is not necessarily right initially if the parent
-    # process was started from a different directory to the notebook.
-    cd(dirname(path))
+    # We check the `execute-dir` key in the options,
+    if haskey(options, "project") && haskey(options["project"], "execute-dir")
+        ed = options["project"]["execute-dir"]
+        if ed == "directory"
+            cd(dirname(path))
+        elseif ed == "project"
```

Comment:
I only see a description of `project` here, https://quarto.org/docs/projects/code-execution.html#working-dir, not `directory`. Is that documented elsewhere. It also doesn't mention that you can pass a path, is that described somewhere else?

Reply:
I just looked this up - Quarto's project file schema defines `execute-dir` as an `enum: [file, project]`, so we should probably only support those two arguments.

https://github.com/quarto-dev/quarto-cli/blob/6b363d996e31c65826c2e8186c14a5da6019af96/src/resources/schema/project.yml#L16-L24